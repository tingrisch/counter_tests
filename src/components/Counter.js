import React from "react";
import styled from "styled-components";

const CounterDiv = styled.div`
  font-family: Arial;
  font-size: 22pt;
  color: ${props => props.count < 0 ? 'red' : 'green'};
`;

const Counter = ({ count }) =>
  <CounterDiv count={count}>
    {count}
  </CounterDiv>;

export default Counter;
