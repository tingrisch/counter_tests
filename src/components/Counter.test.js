import React from "react";
import Counter from "./Counter";
import renderer from "react-test-renderer";

describe("Counter", () => {
  it("renders without crashing", () => {
    const tree = renderer.create(<Counter />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("displays the correct count passed in props", () => {
    const tree = renderer.create(<Counter count={3} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
