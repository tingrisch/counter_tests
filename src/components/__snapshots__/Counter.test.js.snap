// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter displays the correct count passed in props 1`] = `
<div
  className="sc-bdVaJa jeITAu"
>
  3
</div>
`;

exports[`Counter renders without crashing 1`] = `
<div
  className="sc-bdVaJa jeITAu"
/>
`;
